### PR TITLE
dsp-filters/20170309: Added dsp-filters recipe

### PR DESCRIPTION
**Pull request by klausholstjacobsen**
dsp-filters/20170309

Added recipe

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by RubenRBS — DISMISSED** [review]
Thanks a lot! I've pushed some cleanup code, but the rest of the code looks great! :)

File: recipes/dsp-filters/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+sources:
+  "20170309":
```
Comment (czoido) [inline recipes/dsp-filters/all/conandata.yml:2]: Thanks a lot for the contribution. One thing about the version is that we tipically add the cci. prefix to distinguish as a virtual version provided by CCI (please check https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/conanfile_attributes.md#version)

```suggestion
  "cci.20170309":
```

File: recipes/dsp-filters/all/conandata.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+sources:
+  "cci.20170309":
+    url: "https://github.com/vinniefalco/DSPFilters/archive/acc49170e79a94fcb9c04b8a2116e9f8dffd1c7d.tar.gz"
+    sha256: "330c3960e55b8cdbf041d8ec4790a21a326dc331e43bf14c144ae5d88a47d53d"
+patches:
+  "20170309":
```
Comment (czoido) [inline recipes/dsp-filters/all/conandata.yml:6]:
```suggestion
  "cci.20170309":
```

File: recipes/dsp-filters/config.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+versions:
+  "20170309":
```
Comment (czoido) [inline recipes/dsp-filters/config.yml:2]:
```suggestion
  "cci.20170309":
```

File: recipes/dsp-filters/all/conandata.yml (hunk 1 above)
Comment (klausholstjacobsen) [reply recipes/dsp-filters/all/conandata.yml]: Ahhhhhhhhh. I have actually been looking for information about whether I should prefix with "cci." as I have observed with other packages. Thanks for the link!

**Review by czoido — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

File: recipes/dsp-filters/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        
+        cmake = CMake(self)
+        cmake.configure(build_script_folder=os.path.join(self.source_folder, "shared"))
+        cmake.build()
+
+    def package(self):
+        copy(self, "lib*", src=os.path.join(self.build_folder, "DSPFilters"), dst=os.path.join(self.package_folder, "lib"), keep_path=False)
+        copy(self, "DSPFilters.lib", src=os.path.join(self.build_folder, "DSPFilters", f"{self.settings.build_type}"), dst=os.path.join(self.package_folder, "lib"), keep_path=False)
+
+        copy(self, "*.h", src=os.path.join(self.source_folder, "shared", "DSPFilters", "include"), dst=os.path.join(self.package_folder, "include"))
```
Comment (SpaceIm) [inline recipes/dsp-filters/all/conanfile.py:66]: A patch would have been neater I think, in order to add an install target in upstream CMakeLists.

Comment (klausholstjacobsen) [reply recipes/dsp-filters/all/conanfile.py:66]: I have tried to add fixes upstream. No reaction from the author!

Comment (SpaceIm) [reply recipes/dsp-filters/all/conanfile.py:66]: I mean a patch provided by recipe itself. This library doesn't seem to be maintained anymore, latest commit is from 2017.

**Pull request merged**